core/src/main/java/org/mskcc/cbio/portal/util/ExtendedMutationUtil.java

```diff
@@ -354,4 +360,99 @@ public static ExtendedMutation newMutation()
 
 		return mutation;
 	}
+
+    private static Map<String, Integer> annotateProteinChange(String proteinChange) {
```

Comment:
@inodb @zhx828 we have a `ProteinChangeResolver` this class in `GenomeNexus`: https://github.com/genome-nexus/genome-nexus/blob/67a3ebd56f953a8c3e093e5f909eba4b00a9ee04/service/src/main/java/org/cbioportal/genome_nexus/service/annotation/ProteinChangeResolver.java

It might be possible to merge these two implementations within `GenomeNexus`.

Reply:
This is an issue in the importer. The problem happened here is that the study was annotated long time ago, and the mutation didn't get any protein start and protein end but only protein change. If you reannotate the study, the mutation will be shown with correct protein start. This pull request is trying to fix the issue for the studies without proper annotation.

Reply:
yeah i think it makes sense to merge this logic. In the proteinchangeresolver class it also tries to determine start/end based on protein change. So we have 3 places now where this is done, genome-nexus, genome-nexus-annotation-pipeline and cbioportal importer. Let's follow up later and make a single utility function we can import